### ISSUE TITLE
Clarification on ZenDNN tuning for tensorflow

### ISSUE THREAD

**Issue by sagi-scalers-ai** [amd/ZenDNN#9]
We are refering the below ZenDNN tuning guide for Tensorflow models.
[Tuning guide](https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/tuning-guides/58205-epyc-9004-tg-aiml.pdf#page=33&zoom=100,46,462)

In the section 9.2, For Resnet50 it is recommended to use below settings.
```
export TF_ENABLE_ONEDNN_OPTS=0
export ZENDNN_GEMM_ALGO=3
export ZENDNN_ENABLE_MEMPOOL=2
export ZENDNN_TENSOR_POOL_LIMIT=1024
export GOMP_CPU_AFFINITY=0-95
export OMP_NUM_THREADS=96
```

My question is regarding `TF_ENABLE_ONEDNN_OPTS=0` setting which is ment to disable ZenDNN. Can you clarify on this if it is expexted to disable ZenDNN for Tensorflow models.

**Comment by dineshchitlangia** [comment]
@sagi-scalers-ai This is incorrect. We will get the document updated. Thanks for bringing this to our attention.

The correct setting is:
`export TF_ENABLE_ZENDNN_OPTS=1`

**Comment by ratan-prasad** [comment]
Hi, these are the recommended setting for Resnet50:

export ZENDNN_CONV_ALGO=3
export ZENDNN_TENSOR_POOL_LIMIT=512
export ZENDNN_TF_CONV_ADD_FUSION_SAFE=1
export TF_ENABLE_ZENDNN_OPTS=1
export OMP_NUM_THREADS=96
export GOMP_CPU_AFFINITY=0-95

Please note,  when TF_ENABLE_ZENDNN_OPTS is set to 1, TF_ENABLE_ONEDNN_OPTS is set to 0 inside TensorFlow code.
Above setting works for ZenDNN v4.0 and ZenDNN v4.1 release.

**Comment by yru-weighed** [comment]
title: In Control for Commercial Payments
---

In Control for Commercial Payments (ICCP) streamlines business transactions using configurable spending controls and enhanced data.

# Overview

ICCP allows you to create and manage virtual card numbers (VCNs) that offer a flexible, simple, and secure way to pay, while capturing the data you need to reconcile payments later.

mastercard

### Unique card number for every transaction

A new account number is generated for each transaction and used only once, which gives each purchase its own unique identifier

controls

### Set transaction level controls

Transaction-level authorization lets you define how, where, and when cards are used

data-repository\_wht

### Get enhanced data

Information important to each individual payment is included in each transaction

gears

### Automated reconciliation

VCN data is available for reconciliation reporting

processing

### Workflow control

Approval routing for transactions follows the existing approval process - can also be used as a key tool to reinvent your requisition-to-pay process

## Requesting a Virtual Card Number

To create a VCN using ICCP you submit a purchase request. The purchase request captures important details about the request, including real card details, supplier details, and the controls that constrain the card usage.

## Adding Invoice Details

ICCP allows you to attach invoice details to a purchase request. For each virtual card, you can include up to 5000 invoice details. The invoice details are included in the email that is sent to the supplier.

## Requesting Reports

You can generate reports of the authorization and clearing activities that occur on a VCN and use these reports to reconcile payments and find discrepancies.

# How It Works

## Initial Setup

To use the ICCP API to issue and manage virtual cards, you must be one of the following:

*   An issuer of Mastercard commercial credit cards
*   A corporate customer of such an issuer
*   A payment agent acting on behalf of the issuer or its corporate customers
